File: java/arrays/IteratingOverArraysAndUpdatingElements.java
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

class IteratingOverArraysAndUpdatingElements {
  public static void main(String[] args) {
    int[] array = {1, 3, 5};

    System.out.println("Before:");
    for (int i = 0; i < array.length; i++)
      System.out.println(String.format("array[%d]: %d", i, array[i]));

    for (int i = 0; i < array.length; i++)
      array[i] *= 2;

    System.out.println("After:");
    for (int i = 0; i < array.length; i++)
      System.out.println(String.format("array[%d]: %d", i, array[i]));

    System.out.println();
  }
}

File: c++/arrays/iterating_over_arrays_and_updating_elements.cpp
#include <iostream>

int main()
{
  int array[3] = {1, 3, 5};

  std::cout << "Before:\n";
  for (int i = 0; i < 3; i++)
    std::cout << "[" << i << "] => " << array[i] << std::endl;

  for (int i = 0; i < 3; i++)
    array[i] *= 2;

  std::cout << "After:\n";
  for (int i = 0; i < 3; i++)
    std::cout << "[" << i << "] => " << array[i] << std::endl;

  std::cout << std::endl;

  array[0] = 1;
  array[1] = 3;
  array[2] = 5;

  std::cout << "Before:\n";
  for (int i = 0; i < 3; i++)
    std::cout << "[" << i << "] => " << array[i] << std::endl;

  for (int i = 0; i < 3; i++)
    *(array + i) *= 3;

  std::cout << "After:\n";
  for (int i = 0; i < 3; i++)
    std::cout << "[" << i << "] => " << array[i] << std::endl;

  std::cout << std::endl;

  array[0] = 1;
  array[1] = 3;
  array[2] = 5;

  std::cout << "Before:\n";
  for (int i = 0; i < 3; i++)
    std::cout << "[" << i << "] => " << array[i] << std::endl;

  for (int i = 0, *element_pointer = array; i < 3; i++)
    *(element_pointer++) *= 4;

  std::cout << "After:\n";
  for (int i = 0; i < 3; i++)
    std::cout << "[" << i << "] => " << array[i] << std::endl;

  std::cout << std::endl;

  array[0] = 1;
  array[1] = 3;
  array[2] = 5;

  std::cout << "Before:\n";
  for (int i = 0; i < 3; i++)
    std::cout << "[" << i << "] => " << array[i] << std::endl;

  for (int &element : array)
    element *= 5;

  std::cout << "After:\n";
  for (int i = 0; i < 3; i++)
    std::cout << "[" << i << "] => " << array[i] << std::endl;

  std::cout << std::endl;
}
